### PR TITLE
Add pre-req checker script and associated documentation

### LINKED ISSUE #886: Write prerequisites section for admin tutorial
> estimate 2

Most documentation is covered top level README, this work is primarily about writing the test script.

Script below may be useful.
https://github.com/geopm/endeavor-tutorial/blob/master/test-msr-safe.sh

Example RAPL script in geopm/story-806

Parent story #806

Done Criteria:
1. Verify documentation and elaborate on pointers in this doc.
2. Write the script to configure as much as possible in the system.
3. Write the test script to verify the system is configured properly.

### PR DESCRIPTION
**Pull request by bgeltz**
Part of story #806.

- Fixes #886.

### CODE REVIEW

File: tutorial/admin/README.md [hunk 1]
```diff
@@ -12,7 +12,15 @@ The directory contains scripts that can be used to configure systems
 for use with the GEOPM runtime.  The scripts associated with each
 section are numbered accordingly.
 
-0. Set up a system to use the GEOPM static policy plugin for SLURM
+0. Verify that the required pre-requisites for running GEOPM are met
+--------------------------------------------------------------------
+Run ```./test_prereqs.sh``` on a compute node in the context of a regular user
+to ensure the GEOPM components are installed and functioning properly.
+In the event that somthing is misconfigured, an ERROR or WARNING message
```
Comment (dianarg) [inline tutorial/admin/README.md]: somthing->something

File: tutorial/admin/check_whitelist.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+RC=0
+
+FILE=/dev/cpu/msr_whitelist
+if [ ! -c ${FILE} ]; then
+    echo "ERROR: ${FILE} is not available.  Please install msr-safe."
+    RC=1
+else
+    WHITELIST=$(cat ${FILE}) # This may require root.  If it does, re-run this script as root, or create a temp file
```
Comment (dianarg) [inline tutorial/admin/check_whitelist.sh:51]: I think the msr whitelist still needs to be user readable, right?

File: tutorial/admin/test_prereqs.sh [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+command_available_and_executable geopmanalysis
+command_available_and_executable geopmread
+command_available_and_executable geopmwrite
+command_available_and_executable geopmadmin
+
+# Check GNU vs. Intel Toolchain
+if ! ldd $(which geopmctl) | grep --quiet libimf; then
+    echo "WARNING: Please build geopm with the Intel Toolchain"
+    echo "         to ensure the best performance."
+fi
```
Comment (dianarg) [inline tutorial/admin/test_prereqs.sh]: maybe geopmbench?

File: tutorial/admin/test_prereqs.sh [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+        RC=1
+    elif [ "$(grep --quiet ^#RemoveIPC ${CONF} | cut -d= -f2)" = "yes" ]; then
+        echo "ERROR: The compile time defaults for systemd will cause issues"
+        echo "       with GEOPM.  Set RemoveIPC=no in ${CONF}"
+        echo "       Please read the \"SYSTEMD CONFIGURATION\" section of the main README."
+        RC=1
+    elif ! grep --quiet RemoveIPC ${CONF}; then
+        echo "WARNING: Unable to determine RemoveIPC value from ${CONF}"
+        echo "         Run the GEOPM unit tests to ensure there are no issues"
+        echo "         with shared memory.  SharedMemoryTest.fd_check"
```
Comment (dianarg) [inline tutorial/admin/test_prereqs.sh]: looks like some words missing

File: tutorial/admin/test_prereqs.sh [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+fi
+
+FILE=/dev/cpu/msr_batch
+if [ ! -c ${FILE} ]; then
+    echo "ERROR: ${FILE} is not available.  Please install msr-safe."
+    RC=1
+fi
+
+if ! ${MSRSAVE} /tmp/msrsave_test; then
+    echo "ERROR: ${MSRSAVE} had a non-zero return code."
```
Comment (dianarg) [inline tutorial/admin/test_prereqs.sh]: msrsave unit test failed.  What are they expected to do?   probably should contact GEOPM or msr-safe maintainers

Comment (dianarg) [reply tutorial/admin/test_prereqs.sh]: This should also check for existence of msrsave executable

File: tutorial/admin/test_prereqs.sh [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+    echo "ERROR: Unable to set frequency with geopmwrite."
+    RC=1
+else
+    CURRENT_FREQ=$(geopmread MSR::PERF_CTL:FREQ package 0)
+    if [ "${CURRENT_FREQ}" != "${TEST_FREQ}" ]; then
+        echo "ERROR: Current frequency (${CURRENT_FREQ}) != test frequency (${TEST_FREQ})."
+        RC=1
+    fi
+fi
+
```
Comment (dianarg) [inline tutorial/admin/test_prereqs.sh:178]: might want to save and restore frequency

File: tutorial/admin/check_whitelist.sh [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+                    break
+                fi
+            fi
+        done
+        if [ ${FOUND} -ne 1 ]; then
+            echo "ERROR: Required MSR ${GEOPM_MSRS[idx]} was not found in the whitelist."
+            RC=1
+        fi
+    done
+fi
```
Comment (dianarg) [inline tutorial/admin/check_whitelist.sh:78]: maybe a comment for the non-guru bash users

Comment (dianarg) [reply tutorial/admin/check_whitelist.sh:78]: alternatively, just describe the whole script at the top of the file

**Review by dianarg — CHANGES_REQUESTED** [review]

File: tutorial/admin/check_whitelist.sh (hunk 2 above)
Comment (bgeltz) [reply tutorial/admin/check_whitelist.sh:51]: Yep, the whitelist needs to be user readable for this to function properly as a regular user.  If it is not, they have 3 options:
1. Re-run the check script as root assuming geopmadmin is available in the root user's context.
2. Make the whitelist user readable.
3. sudo cat the whitelist into a temp file that the user has access to and modify the check script.

File: tutorial/admin/test_prereqs.sh (hunk 5 above)
Comment (bgeltz) [reply tutorial/admin/test_prereqs.sh]: The existence check is on line 91.

**Pull request merged**